src/ambianic/pipeline/ai/fall_detect.py

```diff
@@ -0,0 +1,110 @@
+        """Detect objects in sample image."""
+        log.debug("%s received new sample", self.__class__.__name__)
+        if not sample:
+            # pass through empty samples to next element
+            yield None
+        else:
+            try:
+                image = sample['image']
+                inference_result = self.fall_detect(image=image)
+                inf_meta = {
+                    'display': 'Fall Detection',
+                }
+                # pass on the results to the next connected pipe element
+                processed_sample = {
+                    'image': image,
+                    'inference_result': inference_result,
+                    'inference_meta': inf_meta
+                    }
+                yield processed_sample
+            except Exception as e:
+                log.error('Error "%s" while processing sample. '
+                          'Dropping sample: %s',
+                          str(e),
+                          str(sample)
+                          )
+                log.warning(stacktrace())
+
+    def calculate_angle(self, p):
+        x1, y1 = p[0][0][0], p[0][0][1]
+        x2, y2 = p[0][1][0], p[0][1][1]
+        x3, y3 = p[1][0][0], p[1][0][1]
+        x4, y4 = p[1][1][0], p[1][1][1]
+        
+        m1 = (y1-y2)/(x1-x2)
+        m2 = (y3-y4)/(x3-x4)
+        
+        theta1 = math.degrees(math.atan(m1))
+        theta2 = math.degrees(math.atan(m2))
+            
+        theta = abs(theta1-theta2)
+        return theta
+
+    def fall_detect(self, image=None):
+        assert image
+        log.debug("Calling TF engine for inference")
+        
+        # Detection using tensorflow posenet module
+        poses = self._pose_engine.DetectPosesInImage(image)
+        
+        inference_result = []
+        pose_vals_list = [[], []]      # [[left shoulder, left hip], [right shoulder, right hip]]
+
+        for i, pose in enumerate(poses):
+            if pose.score < 0.5:
+                continue
+            
+            for label, keypoint in pose.keypoints.items():
+                if (label == 'left shoulder' or label == 'left hip') and (keypoint.score > 0.5):
+                    pose_vals_list[0].append((keypoint.yx[0], keypoint.yx[1]))
+                elif label == 'right shoulder' or label == 'right hip':
+                    pose_vals_list[1].append((keypoint.yx[0], keypoint.yx[1]))
+
+            if not self._prev_vals:
+                res = False
+            else:
+                temp_left_point = [[self._prev_vals[0][0], self._prev_vals[0][1]], [pose_vals_list[0][0], pose_vals_list[0][1]]]
+                left_angle = self.calculate_angle(temp_left_point)
+
+                temp_right_point = [[self._prev_vals[1][0], self._prev_vals[1][1]], [pose_vals_list[1][0], pose_vals_list[1][1]]]
+                rigth_angle = self.calculate_angle(temp_right_point)
+
+                if left_angle > self._fall_factor or rigth_angle > self._fall_factor:
```

Comment:
@bhavikapanara since fall factor is preset to 20 degrees, the test here checks whether the body line is tilted 20' off vertical position. Is that correct? 

I see in the test that you've chosen a person sitting straight up and another person who is crouching and maybe about to fall, but not on the floor. From the image it is not clear if the person will fall or squat back up.

Should we pick a bigger angle, maybe 60 to have a reasonably high level of confidence that its a fall.

I understand that if the angle is bigger, the pose model may not detect correctly the body points, which is why we discussed the approach of rotating the image +/- 90' in such cases to see if you can get better pose estimates.

In a follow up project (not this PR), we will discuss re-training the model with more data using rotation during training so that the model is more aware of fallen positions and won't require rotation in the inference stage later on.

Looking forward to your thoughts.